### ISSUE TITLE
[BUG]: `DataCollatorForSupervisedDataset` in sft_dataset.py not support left padding

### ISSUE THREAD

**Issue by Desperado-Jia** [hpcaitech/ColossalAI#4193]
### 🐛 Describe the bug

    def __call__(self, instances: Sequence[Dict]) -> Dict[str, torch.Tensor]:
        input_ids, labels = tuple([instance[key] for instance in instances] for key in ("input_ids", "labels"))
        input_ids = torch.nn.utils.rnn.pad_sequence(input_ids,
                                                    batch_first=True,
                                                    padding_value=self.tokenizer.pad_token_id)
        labels = torch.nn.utils.rnn.pad_sequence(labels, batch_first=True, padding_value=IGNORE_INDEX)
        return dict(
            input_ids=input_ids,
            labels=labels,
            attention_mask=input_ids.ne(self.tokenizer.pad_token_id),
        )

This function not support left padding (e.g., BLOOM)
